crc32_p.h

```diff
@@ -0,0 +1,19 @@
+#ifndef CRC32_P_H_
+#define CRC32_P_H_
+
+#define GF2_DIM 32      /* dimension of GF(2) vectors (length of CRC) */
+
+
+static uint32_t gf2_matrix_times(const uint32_t *mat, uint32_t vec) {
+    uint32_t sum = 0;
+    while (vec) {
+        if (vec & 1)
+            sum ^= *mat;
+        vec >>= 1;
+        mat++;
+    }
+    return sum;
+}
```

Comment:
Only inline functions should be defined in header files

Reply:
I'd say static functions are also fine if there are legit reasons for it to be there, but I've made it inline now.

Reply:
The problem with non-inline functions in header files is that gcc might inline it anyways but some compilers will make duplicate copies if more than one compilation unit includes the header.

Reply:
IIRC static in that context means the compiler should make separate copies per compilation unit, so therefore that is as expected. It is not generally good practice of course, but in certain cases it can be necessary to avoid making it a global function in a separate compilation unit/source file.
But inlining this one is fine with me. :)

Reply:
Making simple functions explicitly inline lessen the difference between compiler behaviours. It's true that static functions are local to the compilation unit they are included in, but the issue here was not being local or not, the issue was that different compilers treat non-inline static functions in header files differently.